### DISCUSSION TITLE
How to get null-ls to automatically mount to file types

### DISCUSSION THREAD

**Discussion by irishgreencitrus** [mason-org/mason.nvim#385] [Q&A] [answered]
Hey, apologies if this is already a question, however I have looked around and there seems to be no sources with my specific problem.

The problem is that `null-ls` detects all the formatters and linters yet fails to mount to the correct file type.
Is this something on mason's end or should I be looking around configuring null-ls

Any help appreciated

**Comment by AlixShahid** [comment, marked as answer]
you have to explicitly configure each formatter/linters 
for example: to setup clang_format
```lua
require("null-ls").setup({
    sources = {
        require("null-ls").builtins.formatting.clang_format.with({
            command = "clang-format",
            args = {
                "-assume-filename",
                "$FILENAME",
            },
            to_stdin = true,
        })
    },
})

```
[readmore](https://github.com/jose-elias-alvarez/null-ls.nvim#setup)
[more](https://github.com/jose-elias-alvarez/null-ls.nvim/tree/main/lua/null-ls/builtins)
[for ref](https://github.com/Abstract-IDE/Abstract/blob/main/lua/plugins/null-ls_nvim.lua)
